common-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/common/client/json/JSONUtils.java

```diff
@@ -68,4 +69,83 @@ public static JSONValue parseJSON(String jsonStr) throws JSONException{
             throw new JSONException(message, t);
         }
     }
+    
+    
+    /**
+     * @param throwable a serialized JSON Exception
+     * @return the value of the 'errorMessage' key
+     */
+    public static String getJsonErrorMessage(Throwable throwable) {
+        String msg = throwable.getMessage();
+        return getJsonErrorMessage(msg);
+    }
+
+    /**
+     * @param throwable a serialized JSON Exception
+     * @return the value of the 'httpErrorCode' key, or -1
+     */
+    public static int getJsonErrorCode(Throwable throwable) {
+        String msg = throwable.getMessage();
+        return getJsonErrorCode(msg);
+    }
+
+    /**
+     * @param str String representation of a serialized JSON Exception
+     * @return the value of the 'errorMessage' key
+     */
+    public static String getJsonErrorMessage(String str) {
+        try {
+            JSONObject exc = JSONParser.parseStrict(str).isObject();
+            if (exc != null && exc.containsKey("errorMessage")) {
+                return retrieveErrorMessage(exc);
+            } else if (exc != null && exc.containsKey("exception")) {
+                JSONObject nestedExc = exc.get("exception").isObject();
+                return retrieveErrorMessage(nestedExc);
+            }
+        } catch (Exception e) {
+            if (str != null) {
+                return str;
+            } else {
+                return "<no reason>";
```

Comment:
It is better if "<no reason>" is a private final constant. 
Also the code can be refactor like this : 
return str != null ? str : NO_REASON;